sdk/python/kfp/_local_client.py

```diff
@@ -0,0 +1,503 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import json
+import logging
+import re
+import subprocess
+from collections import deque
+from typing import Any, Callable, Dict, List, Mapping, Tuple, Union, cast
+
+from . import dsl
```

Comment:
`from kfp import dsl`

Reply:
This might add circular import references which are problematic in Python.
You `import kfp` which starts importing `kfp/__init__.py` which starts importing  `kfp/_local_client.py` which does `import kfp` again.